helix-core/src/main/java/org/apache/helix/controller/dataproviders/BaseControllerDataProvider.java

```diff
@@ -544,6 +544,13 @@ public synchronized void setLiveInstances(List<LiveInstance> liveInstances) {
     return _instanceMessagesCache.getMessages(instanceName);
   }
 
+  public void setStaleMessages(Map<String, Map<String, Message>> staleMessageMap) {
```

Comment:
This is never called? Or there are some other changes that you haven't pushed?

Reply:
Yeah, it's never called. Originally I used it, but then I simplified the code, and it's not used any more. I've removed.